src/main/java/net/javadiscord/javabot/systems/qotw/submissions/SubmissionManager.java

```diff
@@ -91,68 +104,116 @@ public WebhookMessageCreateAction<?> handleSubmission(@NotNull ButtonInteraction
 					}
-					qotwSubmissionRepository.deleteSubmission(thread.getIdLong());
-					thread.delete().queue();
-				}
-			} catch (DataAccessException e) {
-				ExceptionLogger.capture(e, SubmissionManager.class.getSimpleName());
-			}
-		});
+				}));
+	}
+
+	private void getOrRetrieveAuthor(@NotNull QOTWSubmission submission, Consumer<User> onSuccess) {
+		if (submission.hasAuthor()) {
+			onSuccess.accept(submission.getAuthor());
+		} else {
+			submission.retrieveAuthor(author -> {
+				submission.setAuthor(author);
+				submissionCache.put(submission.getThread().getIdLong(), submission);
+				onSuccess.accept(author);
+			});
+		}
+	}
+
+	private QOTWSubmission getOrRetrieveSubmission(@NotNull ThreadChannel thread) {
+		if (submissionCache.containsKey(thread.getIdLong())) {
+			return submissionCache.get(thread.getIdLong());
+		} else {
+			QOTWSubmission submission = new QOTWSubmission(thread);
+			submissionCache.put(thread.getIdLong(), submission);
+			return submission;
+		}
 	}
 
 	private boolean canCreateSubmissions(Member member) {
 		if (member == null) return false;
 		if (member.getUser().isBot() || member.getUser().isSystem()) return false;
-		if (member.isTimedOut() || member.isPending()) return false;
-		return !hasActiveSubmissionThreads(member.getIdLong());
+		return !member.isTimedOut() && !member.isPending();
 	}
 
 	/**
-	 * Checks if the given user has unreviewed submissions.
+	 * Accepts a submission.
 	 *
-	 * @param authorId The user's id.
-	 * @return Whether the user hat unreviewed submissions or not.
+	 * @param hook       The {@link net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent} that was fired.
+	 * @param thread     The submission's {@link ThreadChannel}.
+	 * @param author     The submissions' author.
+	 * @param bestAnswer Whether the submission is among the best answers for this week.
 	 */
-	public boolean hasActiveSubmissionThreads(long authorId) {
-		try (Connection con = dbHelper.getDataSource().getConnection()) {
-			return !qotwSubmissionRepository.getUnreviewedSubmissions(authorId).isEmpty();
-		} catch (SQLException e) {
-			ExceptionLogger.capture(e, getClass().getSimpleName());
-			return false;
+	public void acceptSubmission(InteractionHook hook, @NotNull ThreadChannel thread, @NotNull User author, boolean bestAnswer) {
+		thread.getManager().setName(SUBMISSION_ACCEPTED + thread.getName().substring(1)).queue();
+		pointsService.increment(author.getIdLong());
+		notificationService.withQOTW(thread.getGuild(), author).sendAccountIncrementedNotification();
+		Responses.success(hook, "Submission Accepted",
+				"Successfully accepted submission by " + author.getAsMention()).queue();
+		notificationService.withQOTW(thread.getGuild()).sendSubmissionActionNotification(author, getOrRetrieveSubmission(thread), bestAnswer ? SubmissionStatus.ACCEPT_BEST : SubmissionStatus.ACCEPT);
+		Optional<ThreadChannel> newestPostOptional = config.getSubmissionsForumChannel().getThreadChannels()
```

Comment:
Why does it check for the newest post instead of just creating it there (and not before)?

Reply:
cuz the inital qotw is already created by the bot